nncf/pruning/filter_pruning/algo.py

```diff
@@ -94,10 +96,21 @@ def __init__(self, target_model: NNCFNetwork,
         self.current_flops = self.full_flops
 
         self.weights_normalizer = tensor_l2_normalizer  # for all weights in common case
-        self.filter_importance = FILTER_IMPORTANCE_FUNCTIONS.get(params.get('weight_importance', 'L2'))
+        self.filter_importance = FILTER_IMPORTANCE_FUNCTIONS.get(params.get('filter_importance', 'L2'))
+        self.weight_importance = params.get('weight_importance', 'uniform')
         self.all_weights = params.get("all_weights", False)
         scheduler_cls = PRUNING_SCHEDULERS.get(params.get("schedule", "baseline"))
 
+        if self.weight_importance == 'legr':
```

Comment:
Consider triggering LeGR search in a separate controller method, not during its initialization. This also would help resolve issues with parallelization (the API call can be executed after the model is wrapped with DP/DDP).

Reply:
As discussed, it was decided to leave LeGR inside of initialization with special step of wrapping with DP/DDP.